### PR TITLE
ADD: Annotation cache usage docs

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules/prepare_cache.config [hunk 1]
```diff
@@ -18,8 +18,8 @@ process {
     withName: 'SNPEFF_DOWNLOAD' {
         ext.when         = { params.tools && (params.tools.split(',').contains('snpeff') || params.tools.split(',').contains('merge')) }
         publishDir       = [
-                mode: params.publish_dir_mode,
-                path: { "${params.outdir}/cache/" }
+            mode: params.publish_dir_mode,
+            path: { params.outdir_cache ? "${params.outdir_cache}/": "${params.outdir}/cache/" }
```
Comment (FriederikeHanssen) [inline conf/modules/prepare_cache.config:22]: this is trippy read :D

File: docs/usage.md [hunk 2]
```diff
@@ -779,24 +779,116 @@ For GATK.GRCh38 the links for each reference file and the corresponding processe
...
+By default all is specified in the [igenomes.config](https://github.com/nf-core/sarek/blob/master/conf/igenomes.config) file.
+Explanation can be found for all params in the documentation:
+
+- [snpeff_db](https://nf-co.re/sarek/latest/parameters#snpeff_db)
+- [snpeff_genome](https://nf-co.re/sarek/latest/parameters#snpeff_genome)
+- [vep_genome](https://nf-co.re/sarek/latest/parameters#vep_genome)
+- [vep_species](https://nf-co.re/sarek/latest/parameters#vep_species)
+- [vep_cache_version](https://nf-co.re/sarek/latest/parameters#vep_cache_version)
+
+Whith the previous example of `GRCh38`, these are the values that were used for these params:
```
Comment (FriederikeHanssen) [inline docs/usage.md:840]:
```suggestion
With the previous example of `GRCh38`, these are the values that were used for these params:
```

File: docs/usage.md [hunk 3]
```diff
@@ -779,24 +779,116 @@ For GATK.GRCh38 the links for each reference file and the corresponding processe
...
+
+```bash
+snpeff_db         = 'GRCh38.105'
+snpeff_genome     = 'GRCh38'
+vep_genome        = 'GRCh38'
+vep_species       = 'homo_sapiens'
+vep_cache_version = '106'
```
Comment (FriederikeHanssen) [inline docs/usage.md:847]: here it is not really clear to me how you can combine the db and the cache and when and how they need to be of the same version

Comment (maxulysse) [reply docs/usage.md:847]: `snpeff_db` and `vep_cache_version`?

Comment (FriederikeHanssen) [reply docs/usage.md:847]: yes, too many versions for 1 tool 😆

Comment (FriederikeHanssen) [reply docs/usage.md:847]: I can't say I am less confused 😆 , but lets merge and improve as we go

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**